### ISSUE TITLE
Spawning process for every parallel request results in high CPU usage

### ISSUE THREAD

**Issue by richardscarrott** [stripe/stripe-node#1202]
So this issue was [raised a while back](https://github.com/stripe/stripe-node/issues/313) but there wasn't any reproduction steps which I guess is why it was closed.

I was profiling our node server and noticed from cold one of our APIs was spending a lot of time in the stripe SDK.

Here's a contrived way to reproduce the issue:

```js
const paymentIntentIds = new Array(50).fill('pi_123');
await Promise.all(paymentIntentIds.map(id => stripe.paymentIntents.retrieve(id));
```

If run from cold (i.e. without having made a prior call to the SDK), it will spawn a new process for all 50 of the above requests because the [`getClientUserAgent` caching is a little naive](https://github.com/stripe/stripe-node/blob/40eaaab26cc7d28b718daa649d129061e5d8a6e0/lib/stripe.js#L409).

You can see the difference between a cold call:
<img width="1363" alt="Screenshot 2021-07-27 at 18 11 51" src="https://user-images.githubusercontent.com/539309/127207453-be86f96d-1e5f-4623-8954-41cfb2e99392.png">

And a 2nd warm call:
<img width="972" alt="Screenshot 2021-07-27 at 18 12 57" src="https://user-images.githubusercontent.com/539309/127207479-34e19d48-d83e-4dc6-b6f3-e1cc0dcb1797.png">

If you're wondering why one might be making so many parallel requests at once, our use case is fetching a page of orders in our DB which all have a FK reference to a payment intent in stripe (we don't currently replicate the payment intents in our DB) and unlike older stripe resources, there's no batch endpoint for payment intents AFAIK.

I discovered the issue in 7.5.3 but can also reproduce in **v8.165.0**.

**Comment by richardscarrott** [comment]
In the meantime, I've patched our stripe client with:

```js
Stripe.prototype.getClientUserAgent = cb =>
  cb(`HACK: https://github.com/stripe/stripe-node/issues/1202`);
```

Hopefully won't cause any problems with the telemetry on your end?

**Comment by richardm-stripe** [comment]
Hello @richardscarrott, thank you for the report and the great detective work. A fix for this in #1205 has been merged and will be released soon.
